### DISCUSSION TITLE
Script to run multiple ORT steps in one call

### DISCUSSION THREAD

**Discussion by NicolasToussaint** [oss-review-toolkit/ort#9280] [Show and tell]
We (with @woznik) wrote a script to run multiple ORT steps in one call, to make producing ORT reports super easy from CI/CD.

There are 5 hard coded profiles, that will run different steps and exit, automatically reusing previous report file to feed the next step. For example, profile `LICENSING` will run `analysis`, `scan` and `report` steps.

The script embeds other features that we needed.

I guess the "hard coded" part is not ideal, but it wouldn't be hard to read a Json configuration to that anyone could create their own profiles.

Before I go into this, I would love to get feedback to know if that feature would be of any interest as a contribution to the ORT project ?

Current script is here: https://github.com/Orange-OpenSource/ort/blob/main-with-profiles/scripts/run-profiles.sh (with adapted [docker run script](https://github.com/Orange-OpenSource/ort/blob/main-with-profiles/scripts/docker_run-profiles.sh))

**Comment by sschuberth** [comment]
Hi @NicolasToussaint, thanks for sharing what you've done!

To give you an overview, we currently have
- the ORT [helper-cli]()https://github.com/oss-review-toolkit/ort/tree/main/helper-cli (`orth`)
- the `orthw` wrapper for `orth`
  - in the [original shell script version](https://github.com/oss-review-toolkit/orthw)
  - in the [new Python version](https://github.com/oss-review-toolkit/orthw/tree/heliocastro/python_version)
- scripts that come with CI integrations like for [GitLab](https://github.com/oss-review-toolkit/ort-gitlab-ci/tree/main/scripts)

My simple fear is that adding yet more scripts creates too much confusion for our users what to use. Do you see a chance for consolidating your work with any of the above?

**Comment by NicolasToussaint** [comment]
Hi @sschuberth ,

That's great, thanks for the many pointers, we will check them out :+1:

**Comment by sschuberth** [comment]
Any update @NicolasToussaint?

**Comment by NicolasToussaint** [comment]
> Any update @NicolasToussaint?

Hi @sschuberth ,
Sorry, no update on this topic yet
